ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/rules/NoEmptyFileRule.kt

```diff
@@ -0,0 +1,59 @@
+package com.pinterest.ktlint.ruleset.standard.rules
+
+import com.pinterest.ktlint.rule.engine.core.api.ElementType
+import com.pinterest.ktlint.rule.engine.core.api.Rule
+import com.pinterest.ktlint.rule.engine.core.api.editorconfig.EditorConfig
+import com.pinterest.ktlint.rule.engine.core.api.editorconfig.EditorConfigProperty
+import com.pinterest.ktlint.ruleset.standard.StandardRule
+import org.ec4j.core.model.PropertyType
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+
+public class NoEmptyFileRule :
+    StandardRule(id = "no-empty-file", usesEditorConfigProperties = setOf(NO_EMPTY_FILE_PROPERTY)),
+    Rule.Experimental {
+    private var noEmptyFile = NO_EMPTY_FILE_PROPERTY.defaultValue
+
+    override fun beforeFirstNode(editorConfig: EditorConfig) {
+        noEmptyFile = editorConfig[NO_EMPTY_FILE_PROPERTY]
+    }
+
+    override fun beforeVisitChildNodes(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit,
+    ) {
+        if (noEmptyFile) {
+            node
+                .takeIf { it.elementType == ElementType.FILE }
+                ?.takeIf { it.textLength == TEXT_LENGTH_EMPTY_FILE_CONTAINS }
+                ?.let {
+                    val filePath = it.psi.containingFile.virtualFile.name
+                    val fileName =
+                        filePath
+                            .replace("\\", "/") // Ensure compatibility with Windows OS
+                            .substringAfterLast("/")
```

Comment:
What is the reasoning behind showing the filename without path? The path might be relevant in case multiple files with same name exist and not all of them are empty.